source/collections/Vector2D.ooc

```diff
@@ -38,6 +38,52 @@ Vector2D: class <T> {
 	_elementPosition: func (row, column: Int, columnCount := this columnCapacity) -> Int {
 		result := columnCount * row + column
 	}
+	resize: func (newRowCapacity, newColumnCapacity: Int) {
+		temporaryResult: T*
+
+		maximumRowCapacity := this rowCapacity > newRowCapacity ? newRowCapacity : this rowCapacity
+		maximumColumnCapacity := this columnCapacity > newColumnCapacity ? newColumnCapacity : this columnCapacity
+
+		if (newRowCapacity == this rowCapacity && newColumnCapacity == this columnCapacity)
+			return
+
+		temporaryResult = gc_calloc(newRowCapacity * newColumnCapacity, T size)
```

Comment:
If the `newRowCapacity` is the same as `this rowCapacity ` you should be able to use `realloc` directly on the old memory block, without the need to allocate temporary buffer.